### PR TITLE
fix: change logic to separate functions from layers in mock

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #4992 *

*Description of changes:*

* Changed the logic to mock functions with layers when dependency on a function

* Added tests for the same

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1116,9 +1116,9 @@ workflows:
           filters:
             branches:
               only:
-                # - master
+                - master
                 - graphqlschemae2e
-                # - beta
+                - beta
```
Comment (jhockett) [inline .circleci/config.yml:1121]: Is this related to the mock changes?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1161,7 +1161,7 @@ workflows:
             - test
             - mock_e2e_tests
             - graphql_e2e_tests
-            # - integration_test
+            - integration_test
```
Comment (jhockett) [inline .circleci/config.yml:1164]: Is this related to the mock changes?

File: .circleci/config.yml (hunk 1 above)
Comment (akshbhu) [reply .circleci/config.yml:1121]: No , this is because of issues in circleCI yesterday. I am now changing it back to original

File: .circleci/config.yml (hunk 2 above)
Comment (akshbhu) [reply .circleci/config.yml:1164]: same as above

**Comment by prophet1996** [comment]
Hi, @akshbhu just wondering when will this be available on taking npm update
